Automattic/wp-calypso
# Plans 2023: Hide custom domain feature if domain credit has been redeemed

Related to https://github.com/Automattic/martech/issues/2850 and D145279-code

## Proposed Changes

Currently, our pricing grid still shows "Free domain for one year" in the logged-in /plans on a paid site even if its domain credits _have already been used_.

There are users who have mistaken this to mean that they could gain another free domain for a year upon upgrading their plan. Because of this, we should exclude the "Free domain for one year" feature if one has already been redeemed for their site.

## Screenshots
### Before
<img width="1257" alt="Screenshot 2024-04-15 at 1 24 52 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/8bd6dc41-bd9a-4536-b013-9cec17112c23">

<img width="1120" alt="Screenshot 2024-04-15 at 1 52 14 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/5f0515ba-32cd-4afe-801e-9f9162e3f32b">

### After
<img width="1411" alt="Screenshot 2024-04-15 at 1 24 44 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/06368bb7-b468-41b1-958d-b2efc72fb724">

<img width="1111" alt="Screenshot 2024-04-15 at 1 52 31 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/43f856bf-e493-4902-9211-300bdf70238e">

## Testing Instructions



- Test alongside D145279-code
  - Sandbox public-api.wordpress.com 
  - Patch D145279
- Check out this branch
- Set up a new site with a free plan 
- Visit calypso.localhost:3000/plans/{site_slug_for_new_site}
- Verify that we still display the "Free domain for one year" feature in the features grid and the comparison grid
- Upgrade to a plan that provides free domain for a year ( starter, explorer, creator, entrepreneur )
- Revisit calypso.localhost:3000/plans/{site_slug_for_new_site}
- Verify that we no longer display the "Free domain for one year" for plan upgrades or downgrades

## Pre-merge Checklist
